src/dmd/backend/elem.d

```diff
@@ -781,146 +897,149 @@ int el_anydef(elem *ed, elem *e)
-    e->E1 = e1;
-    e->E2 = e2;
+    e.EV.E1 = e1;
+    e.EV.E2 = e2;
     return e;
 }
 
-elem * el_bin(unsigned op,tym_t ty,elem *e1,elem *e2)
+elem * el_bin(uint op,tym_t ty,elem *e1,elem *e2)
 {   elem *e;
 
-#if 0
+static if (0)
+{
     if (!(op < OPMAX && OTbinary(op) && e1 && e2))
-        *(char *)0=0;
-#endif
+        *cast(char *)0=0;
+}
     assert(op < OPMAX && OTbinary(op) && e1 && e2);
-    assert(MARS || !PARSER);
+    version (MARS) { } else assert(!PARSER);
     elem_debug(e1);
     elem_debug(e2);
     e = el_calloc();
-    e->Ety = ty;
-    e->Eoper = op;
-    e->E1 = e1;
-    e->E2 = e2;
+    e.Ety = ty;
+    e.Eoper = cast(ubyte)op;
+    e.EV.E1 = e1;
+    e.EV.E2 = e2;
     if (op == OPcomma && tyaggregate(ty))
-        e->ET = e2->ET;
+        e.ET = e2.ET;
     return e;
 }
 
 /************************
  * Make a unary operator node.
  */
 
-elem * el_unat(unsigned op,type *t,elem *e1)
+elem * el_unat(uint op,type *t,elem *e1)
 {   elem *e;
 
-#ifdef DEBUG
-    if (!(op < OPMAX && OTunary(op) && e1))
-        dbg_printf("op = x%x, e1 = %p\n",op,e1);
-#endif
+    debug if (!(op < OPMAX && OTunary(op) && e1))
+        printf("op = x%x, e1 = %p\n",op,e1);
+
     assert(op < OPMAX && OTunary(op) && e1);
     assert(PARSER);
     elem_debug(e1);
     e = el_calloc();
-    e->Eoper = op;
-    e->E1 = e1;
+    e.Eoper = cast(ubyte)op;
+    e.EV.E1 = e1;
     if (t)
     {
         type_debug(t);
-        t->Tcount++;
-        e->ET = t;
+        t.Tcount++;
+        e.ET = t;
     }
     return e;
 }
 
-elem * el_una(unsigned op,tym_t ty,elem *e1)
+elem * el_una(uint op,tym_t ty,elem *e1)
 {   elem *e;
 
-#ifdef DEBUG
-    if (!(op < OPMAX && OTunary(op) && e1))
-        dbg_printf("op = x%x, e1 = %p\n",op,e1);
-#endif
+    debug if (!(op < OPMAX && OTunary(op) && e1))
+        printf("op = x%x, e1 = %p\n",op,e1);
+
     assert(op < OPMAX && OTunary(op) && e1);
-    assert(MARS || !PARSER);
+    version (MARS) { } else assert(!PARSER);
     elem_debug(e1);
     e = el_calloc();
-    e->Ety = ty;
-    e->Eoper = op;
-    e->E1 = e1;
+    e.Ety = ty;
+    e.Eoper = cast(ubyte)op;
+    e.EV.E1 = e1;
     return e;
 }
 
 /*******************
  * Make a constant node out of integral type.
  */
 
-elem * el_longt(type *t,targ_llong val)
+extern (C) elem * el_longt(type *t,targ_llong val)
 { elem *e;
 
   assert(PARSER);
   e = el_calloc();
-  e->Eoper = OPconst;
-  e->ET = t;
-  if (e->ET)
+  e.Eoper = OPconst;
+  e.ET = t;
+  if (e.ET)
   {     type_debug(t);
-        e->ET->Tcount++;
+        e.ET.Tcount++;
   }
-  e->EV.Vllong = val;
+  e.EV.Vllong = val;
   return e;
 }
 
-extern "C" // necessary because D <=> C++ mangling of "long long" is not consistent across memory models
+extern (C) // necessary because D <=> C++ mangling of "long long" is not consistent across memory models
 {
 elem * el_long(tym_t t,targ_llong val)
 { elem *e;
 
-  assert(MARS || !PARSER);
+    version (MARS)
+    { }
+    else
+        assert(!PARSER);
```

Comment:
The formatting you used before `version (MARS) { } else assert(!PARSER);` looks more natural ;-)